extensions/replication/utils/QueueEntry.js

```diff
@@ -159,6 +163,30 @@ class QueueEntry {
         return location.size;
     }
 
+    // Object metadata may contain multiple elements for a single part if
+    // the part was originally copied from another MPU. Here we reduce the
+    // locations array to a single element for each part.
+    getReducedLocations() {
```

Comment:
I would suggest making this change in another PR as this is out of scope. I think there is more context to this as in it may impact the behavior of the GET Object ?partNumber on the target bucket.

Reply:
This is just the function code moved to a member of QueueEntry and function renamed from `_getLocations` to `_getReducedLocations`. It has unit tests as well (pre-existing ones, moved to QueueEntry.spec.js), so I think it's safe to keep in this PR (actually it would take some time to revert the changes as it's part of the main commit, not sure it's worth it).

Reply:
Ah I see. Let's keep it then.